New walls, doors, and windows

CakeQ opened the pull request:
🆑 Cakey
imageadd: New wall sprites have been added.
imageadd: New door sprites have been added.
imageadd: New window sprites have been added.
imageadd: New grille sprites have been added.
rscadd: Airlocks are now paintable.
rscadd: Window frames have been added, which connect to walls for that aesthetic look.
rscadd: Airlocks can now be painted using the airlock painter.
rscadd: Added paint presets to the paintgun.
maptweak: The Torch has been modified to use window frames instead of just windows.
maptweak: Maintenance has been given more catwalks.
maptweak: Added binoculars to security checkpoints.
/🆑

Window frames will be created from the same material as walls and can have grilles and windows added to them. Empty frames will be able to be used as counters between rooms as well as support future implementation of other frame constructions (Aquariums, display cases, etc).

To-Do: 
- [x] New airlock code.
- [x] Station airlock sprites.
- [x] EVA airlock sprites.
- [x] Secure airlock sprites.
- [x] Vault airlock sprites.
- [x] Hatch airlock sprites.
- [x] Elevator airlock sprites.
- [x] Double airlock sprites.
- [x] Fire door sprites.
- [x] Mineral door sprites.
- [x] Shutter sprites.
- [x] Blast door sprites.
- [x] Ailrock brace sprites.
- [x] New door assembly code.
- [x] Window frame code.
- [x] Window frame construction code.
- [x] Window frame sprites.
- [x] Wall sprites
- [x] Reinforced wall sprites.
- [x] Shuttle wall sprites.
- [x] Cult wall sprites.
- [x] Airlock painter.
- [x] SQUASH

Previews:
![Walls and Windows](https://i.imgur.com/X3svpZn.png)

![Security](https://i.imgur.com/XC7Koko.png)

![Shuttle walls](https://i.imgur.com/xBzWatB.png)

![Elevator](https://i.imgur.com/oTfTF3J.png)

![Secure Doors](https://i.imgur.com/XxtZSKS.png)

![Shutters and Doors](https://i.imgur.com/AAPwYWb.png)

![Grilles](https://i.imgur.com/s1phu1u.png)

![EVA](https://i.imgur.com/AIBfr5o.png)

HeyBanditoz commented:
What in God's name did you do to poor torch-1.dmm?

CakeQ commented:
ah shit, didnt mean to even TOUCH the map files. They were edited for testing

edit: obviously had to replace all glass airlock paths tho

LorenLuke commented:
@CakeQ 
Can you show this with the current wall sprites, bit the new doors and windows, for reference?

comma commented:
Gonna look like ass cause top down vs 3/4

chaoko99 reviewed (commented):
Aren't these *literally* Supernorn's goonwalls? We need to credit them.

chaoko99 commented:
Hm, seems like a waste to get all these walls after the old ones.

chaoko99 commented:
Actually fuck what I said, they look great.

chaoko99 reviewed (commented):
Do these work with client rotation? What about cultist walls?

SilenceInsider commented:
Horrible.

chaoko99 commented:
@SilenceInsider Care to offer insight on how to make it better, rather than "it sux"?

SilenceInsider commented:
Not to make it 3/4?

SilenceInsider commented:
This style literally uses 1/5 chunk of sprite just to display top of the object, for what reason you want to have less details on front view just for it?
There is already so limited place to work with, why make it even more restricted?

FreePrivateer commented:
As opposed to now, when it uses (debateably) most of the sprite is to show off the top. Or side or... something, its a little difficult to tell. The 3/4ths perspective gives the game some much needed depth (makes objects feel like they're in a space rather then just kinda placed on a tile).

Using 1/5th of the sprite to provide perspective is an absolutely fine use for a sprite that exists to pretty much just take up space.

BlueNexus commented:
How do we put things on the north/west/east sides of walls without them looking like shit, with this change? It looks like stuff can only go on the south side of walls.

chaoko99 commented:
NornWalls have a solution for this.

For the record: This is a nornwall
![image](https://user-images.githubusercontent.com/4706052/39363546-8e1279dc-49df-11e8-94f0-1bb6210028a9.png)

They have nice looking directions. And, mind, I don't dislike the verticals we have here, buuuut...
![image](https://user-images.githubusercontent.com/4706052/39363575-a40fbc22-49df-11e8-9deb-905639f8f080.png)
![image](https://user-images.githubusercontent.com/4706052/39363578-a6d525dc-49df-11e8-969b-cbe60d5523c1.png)
![image](https://user-images.githubusercontent.com/4706052/39363581-a88dfade-49df-11e8-85b7-09522df1fad2.png)
![image](https://user-images.githubusercontent.com/4706052/39363586-aa1fcf76-49df-11e8-86e1-14651b927dbd.png)
![image](https://user-images.githubusercontent.com/4706052/39363591-ab4078b0-49df-11e8-97d8-947447d79e5c.png)
![image](https://user-images.githubusercontent.com/4706052/39363594-ac7b99f8-49df-11e8-90c6-340f5bd2226d.png)
![image](https://user-images.githubusercontent.com/4706052/39363596-ae7fc49a-49df-11e8-8faf-f1069b0f7f01.png)
etc etc 
![image](https://user-images.githubusercontent.com/4706052/39363599-b3cff4e2-49df-11e8-97de-808f6abbf413.png)

BlueNexus commented:
I don't see how they solve the problem. North-facing sprites on those walls are still going to look shit, and our map uses a lot of north-facing sprites.

BlueNexus commented:
Also, before you say "do this https://i.imgur.com/aOYpUKq.png" that fixes nothing, because we can't tell what any of the sprites are, as they lack detail. How the hell would we know what the north-facing one is?

BlueNexus commented:
For anyone wondering how this looks in-game (note that some stuff is WIP, like the vertical doors and windows)

![](https://i.imgur.com/acI0ZDt.png)

![](https://i.imgur.com/MKaZegX.png)

chaoko99 commented:
eh, all I can say is that it isn't *as* bad.

CakeQ commented:
![reinforced walls](https://i.imgur.com/c1iJFOX.png)

chaoko99 commented:
This could really use some NornDoors.

CakeQ committed:
New walls, windows, and doors

replaces old glass airlock path

replaces glass door paths

adds wall frames

changes

adds animated overlays to doors

tries blending

finalises airlock overlay system

adds directional sprites

progress

changesd

adjust wall icons/code

adjust wallcode

adds double doors to map

fixes wrong door path

removes 3x doors

changes

fixes

finalises airlocks, adds airlock painter

fixes elevator doors

fixes corruption

fixes camera dir checks

changes boi

final commit

converts away sites to new doors and windows

fixes merge issues

final wallcode adjustment

fixes

changes test case

fixes CI fails

review changes

fixes

fixes wall connections

fixes

fixes

fix mapmerpge

makes bearcat sexy

fixes bearcat mapmerge

bearcat fix

fixes bearcat

removes duplicate wallframe

fixes paintgun

ParadoxonKomplikon commented:
Yeah, I really dislike the sprites presented here. The way horizontal and vertical walls have a completely different width is seriously disruptive, you can only put stuff on north walls, and it seems a pain in the ass to see quickly if a vertical wall is simple or reinforced. 

Norn-like walls are already a sharp improvement over those, even though you still lose the ability to put stuff on southern walls. But I'd rather see them, or our current ones, or our old ones, than the ones I'm seeing on these screenshots.

BlueNexus commented:
The torch is supposed to be compact. Stop complaining about it.

BlueNexus commented:
Also, I agree with @ParadoxonKomplikon, I really dislike those sprites.

Minijar commented:
These actually look pretty decent now the pr's nearing completion. Far better than the current blocky ones.

Space0Cat commented:
Im sorta fine with it, Can you throw us a comparsion with cultist walls

afterthought2 commented on code/game/objects/structures/grille.dm:
```diff
@@ -10,16 +10,45 @@
+			for(var/i = 1 to 4)
+				if(other_connections[i] != "0")
+					I = image(icon, "grille_other_onframe[connections[i]]", dir = 1<<(i-1))
+				else
```
You used << in places like this, hence the travis fail.

PsiOmegaDelta commented on code/__defines/colors.dm:
```diff
@@ -50,6 +50,12 @@
+#define COLOR_SKY_BLUE		   "#5ca1cc"
+#define COLOR_PALE_ORANGE	   "#b88a3b"
+#define COLOR_CIVIE_GREEN	   "#b7f27d"
+#define COLOR_TITANIUM		   "#d1e6e3"
```
Inconsistent indentation format. 
With the exception of `COLOR_HULL` space is used, not tabs.

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/__defines/colors.dm:
```diff
@@ -75,6 +81,11 @@
+#define GLASS_COLOR			   "#74b1ee"
+#define GLASS_COLOR_PHORON	   "#7c3a9a"
+#define GLASS_COLOR_TINTED	   "#222222"
+#define GLASS_COLOR_FROSTED	   "#ffffff"
```
See above.

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/camera/camera.dm:
```diff
@@ -292,6 +292,25 @@
+		var/turf/T = get_step(get_turf(src), turn(src.dir, 180))
+		if(istype(T, /turf/simulated/wall))
+			pixel_x = -10
+			pixel_y = 0
```
To my knowledge cameras never rotate, why is this here?

CakeQ replied:
the turn function is there because the cameras are facing opposite their actual dirs

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -52,6 +62,29 @@
+	//Airlock 2.0 Aesthetics Properties
+	//The variables below determine what color the airlock and decorative stripes will be -Cakey
+	var/paintable = TRUE
+	var/stripable = TRUE
```
We're trying to keep boolean vars down. Use flags instead.

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -71,262 +104,254 @@
+	mineral = "phoron"
+
+/obj/machinery/door/airlock/centcom
+	name = "Airlock"
```
"\improper Airlock"

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -71,262 +104,254 @@
 	var/rad_power = 7.5
 
+/obj/machinery/door/airlock/sandstone
+	name = "Sandstone Airlock"
```
If you really must use capitalized names then
"\improper Sandstone Airlock"

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -71,262 +104,254 @@
+	mineral = "sandstone"
+
+/obj/machinery/door/airlock/phoron
+	name = "Phoron Airlock"
```
"\improper Phoron Airlock"

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -71,262 +104,254 @@
+	locked = 1
+
+/obj/machinery/door/airlock/hatch
+	name = "Airtight Hatch"
```
"\improper Airtight Hatch"

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -582,67 +572,166 @@ About the new airlock wires panel:
+			filling_overlay = fill_file
+	if(stripe_color && !(stripe_color == "none"))
+		stripe_overlay = new(stripe_file)
+		stripe_overlay.Blend(stripe_color, ICON_MULTIPLY)
```
Do not conduct non-cached icon operations under any circumstance.

CakeQ replied:
How would I deal with this instead?

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -582,67 +572,166 @@ About the new airlock wires panel:
+
+	if(door_color && !(door_color == "none"))
+		color_overlay = new(color_file)
+		color_overlay.Blend(door_color, ICON_MULTIPLY)
```
Do not use non-cached icon operations under any circumstance.

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -582,67 +572,166 @@ About the new airlock wires panel:
+		stripe_overlay.Blend(stripe_color, ICON_MULTIPLY)
+		if(!glass)
+			stripe_filling_overlay = new(stripe_fill_file)
+			stripe_filling_overlay.Blend(stripe_color, ICON_MULTIPLY)
```
Do not conduct non-cached icon operations under any circumstance.

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -582,67 +572,166 @@ About the new airlock wires panel:
+	else
+		if(door_color && !(door_color == "none"))
+			filling_overlay = new(color_fill_file)
+			filling_overlay.Blend(door_color, ICON_MULTIPLY)
```
Do not conduct non-cached icon operations under any circumstance.

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/doors/multi_tile.dm:
```diff
@@ -1,7 +1,23 @@
 //Terribly sorry for the code doubling, but things go derpy otherwise.
 /obj/machinery/door/airlock/multi_tile
+	name = "Airlock"
```
"\improper Airlock"

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/doors/multi_tile.dm:
```diff
@@ -1,7 +1,23 @@
+	color_fill_file = 'icons/obj/doors/double/fill_color.dmi'
+	stripe_file = 'icons/obj/doors/double/stripe.dmi'
+	stripe_fill_file = 'icons/obj/doors/double/fill_stripe.dmi'
+	glass_file = 'icons/obj/doors/double/fill_glass.dmi'
```
Just so you know, opaque multi-tile objects will result in visual issues.
Only the main turf will actually block sight.

CakeQ replied:
Yeah I know. Settled with making both glass and metal double airlocks transparent for now.

PsiOmegaDelta commented on code/game/machinery/doors/multi_tile.dm:
```diff
@@ -11,18 +27,158 @@
+/obj/machinery/door/airlock/multi_tile/update_connections(var/propagate = 0)
+	var/dirs = 0
+
+	for(var/direction in GLOB.cardinal)
```
This pretty much looks like copy-paste from the other `/update_connection()` proc.

CakeQ replied:
Needs to be respecified because multi_tiled airlocks need to check tiles 2 spots away instead

PsiOmegaDelta commented on code/game/objects/items/devices/floor_painter.dm:
```diff
@@ -64,9 +86,33 @@
+	if(istype(D))
+		var/choice
+		if(D.paintable && D.stripable)
+			choice = input("What do you wish to paint?") as null|anything in list("Paint","Stripe")
```
`input(user, "What do you wish to paint?")`

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/objects/items/devices/floor_painter.dm:
```diff
@@ -64,9 +86,33 @@
+	if(istype(D))
+		var/choice
+		if(D.paintable && D.stripable)
+			choice = input("What do you wish to paint?") as null|anything in list("Paint","Stripe")
```
After an input you must double-check that the mob is still conscious and able, in range of the targeted airlock, still has the floor painter equipped, etc.

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/objects/items/devices/floor_painter.dm:
```diff
@@ -149,6 +197,19 @@
+	if(usr.incapacitated())
+		return
+	var/new_colour = input(usr, "Choose a colour.", "paintgun", paint_colour) as null|anything in preset_colors
+	if(new_colour && new_colour != paint_colour)
```
Must re-check the pre-conditions after an input(), in this case `usr.incapacitated()` as well as that `src` is still in `usr`

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/objects/structures/grille.dm:
```diff
@@ -231,10 +261,34 @@
+	for(var/obj/O in T)
+		if(istype(O, /obj/structure/wall_frame))
+			success = TRUE
+		if(success)
```
..why just not break right after you set `success`?

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/objects/structures/wall_frame.dm:
```diff
@@ -0,0 +1,143 @@
+	blend_objects = list(/obj/machinery/door) // Objects which to blend with
+	noblend_objects = list(/obj/machinery/door/window)
+
+/obj/structure/wall_frame/New(var/new_loc)
```
```
/obj/structure/wall_frame/Initialize()
	. = ..()
	etc.
```

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/objects/structures/wallframe_spawner.dm:
```diff
@@ -0,0 +1,134 @@
+	// sometimes it's useful to plonk these down and activate them all manually,
+	// once all your ducks are in a row. So if we're already playing, only
+	// auto-activate if this has been put down by a maploader, not a creative admin
+	// see https://github.com/Baystation12/Baystation12/pull/9907#issuecomment-114896669
```
This is definitively copy-paste.

CakeQ replied:
Fixed?

PsiOmegaDelta commented on code/game/objects/structures/wallframe_spawner.dm:
```diff
@@ -0,0 +1,134 @@
+				var/obj/machinery/door/D = O
+				D.update_connections()
+				D.update_icon()
+	return
```
Superfluous return.

CakeQ replied:
Fixed.

PsiOmegaDelta commented on code/game/machinery/camera/camera.dm:
```diff
@@ -292,6 +292,25 @@
+		var/turf/T = get_step(get_turf(src), turn(src.dir, 180))
+		if(istype(T, /turf/simulated/wall))
+			pixel_x = -10
+			pixel_y = 0
```
The south direction is fine?

PsiOmegaDelta commented on code/game/machinery/camera/camera.dm:
```diff
@@ -292,6 +292,25 @@
+		var/turf/T = get_step(get_turf(src), turn(src.dir, 180))
+		if(istype(T, /turf/simulated/wall))
+			pixel_x = -10
+			pixel_y = 0
```
Why are simulated walls not as wide as all other potential objects a camera might find itself against?
What if the relevant wall is de-constructed and replaced with another type of solid turf? Will cameras remain incorrectly shifted?
This goes for any and all other objects you pixel-adjust based on adjacent wall type.

CakeQ replied:
Not sure what you mean by the first question but I've gone ahead and refactored both the lighting and camera pixel shifting chunks to be less duplicate lines.
Also not sure how I should go about fixing the pixel adjustment when walls are deconstructed.

PsiOmegaDelta commented:
You have no idea how many times I thought "Wait.. haven't I already seen this before?"
It's copy-paste galore in here, both within itself and from other parts of the codebase.

PsiOmegaDelta commented:
slavers_base.dmm hasn't been properly cleaned and has reverted to the old map format.

CakeQ commented:
All requested changes have been pushed.

SamHPurp commented:
You madmen. This looks beautiful.
